libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -686,7 +686,7 @@ void AP_Baro::init(void)
 #if APM_BUILD_TYPE(APM_BUILD_ArduSub)
 #if AP_BARO_MS56XX_ENABLED
         ADD_BACKEND(AP_Baro_MS56XX::probe(*this,
-                                          std::move(GET_I2C_DEVICE(_ext_bus, HAL_BARO_MS5837_I2C_ADDR)), AP_Baro_MS56XX::BARO_MS5837));
+                                          std::move(GET_I2C_DEVICE(_ext_bus, HAL_BARO_MS5837_I2C_ADDR)), AP_Baro_MS56XX::BARO_MS5837_30BA));
```

Comment:
```suggestion
                                          std::move(GET_I2C_DEVICE(_ext_bus, HAL_BARO_MS5837_I2C_ADDR)), AP_Baro_MS56XX::BARO_MS5837));
```
Assuming it's letting you keep `BARO_MS5837` as a duplicate option of the 30BA variant (without complaining that it's not covered in the switch), keeping it general at the probing step is more indicative of the actual behaviour.

If it doesn't allow `BARO_MS5837` as an `MS56XX_TYPE` without it being one of the switch cases, then it can be removed entirely and this probe can be set to the 30BA variant, but you'd need to add a comment explaining that it actually probes for the MS5837 generally (not just the 30BA variant), and determines the type during initialisation.